src/client/App.tsx

```diff
@@ -153,87 +153,99 @@ function App() {
-  // TODO make this a URL param
-  // Migrated to useReducer
-  //const [activeFileId, setActiveFileId] =
-  //useState<FileId | null>(null);
-
-  // The ordered list of tabs.
-  // TODO make this a URL param
   // https://react.dev/reference/react/useReducer
-  // const [tabList, setTabList] = useState<Array<FileId>>([]);
   const [state, dispatch] = useReducer(reducer, {
-    submitOperation,
     tabList: [],
-    // activeFileId: null,
-    // theme: defaultTheme,
-    // isSettingsOpen:false
+    activeFileId: null,
+    theme: defaultTheme,
+    isSettingsOpen: false,
   });
 
   //Reducer states
   const tabList = state.tabList;
   const activeFileId = state.activeFileId;
+  const theme = state.theme;
+  const isSettingsOpen = state.isSettingsOpen;
   // TODO phase this out as we complete the refactoring
   // It's here now for backwards compatibility
-  const setTabList = (tabList) => {
-    dispatch({
-      type: 'set_tab_list',
-      tabList,
-    });
-  };
 
-  const setActiveFileId = (activeFileId) => {
-    dispatch({
-      type: 'set_active_fileId',
-      activeFileId,
-    });
-  };
-  const openTab = (fileId: FileId) => {
-    dispatch({
-      type: 'open_tab',
-      fileId,
-    });
-  };
+  const setTabList = useCallback(
+    (tabList) => {
+      dispatch({
+        type: 'set_tab_list',
+        tabList,
+      });
+    },
+    [dispatch],
+  );
 
-  const closeTab = (fileIdToRemove: FileId) => {
-    dispatch({
-      type: 'close_tab',
-      fileIdToRemove,
-    });
-  };
+  const setActiveFileId = useCallback(
+    (activeFileId) => {
+      dispatch({
+        type: 'set_active_fileId',
+        activeFileId,
+      });
+    },
+    [dispatch],
+  );
 
-  const multiCloseTab = (idsToDelete: Array<FileId>) => {
-    dispatch({
-      type: 'multi_close_tab',
-      idsToDelete,
-    });
-  };
-
-  // Logic for opening and closing tabs. Migrated to useReducer
-  /*
-  const {} = useTabsState(
-    activeFileId,
-    setActiveFileId,
-    tabList,
-    setTabList,
-  ); */
-
-  // The current theme.
-  // TODO persist this in local storage
-  const [theme, setTheme] =
-    useState<ThemeLabel>(defaultTheme);
+  const openTab = useCallback(
+    (fileId: FileId) => {
+      dispatch({
+        type: 'open_tab',
+        fileId,
+      });
+    },
+    [dispatch],
+  );
+
+  const closeTab = useCallback(
+    (fileIdToRemove: FileId) => {
+      dispatch({
+        type: 'close_tab',
+        fileIdToRemove,
+      });
+    },
+    [dispatch],
+  );
+
+  const multiCloseTab = useCallback(
+    (idsToDelete: Array<FileId>) => {
+      dispatch({
+        type: 'multi_close_tab',
+        idsToDelete,
+      });
+    },
+    [dispatch],
+  );
+
+  const setTheme = useCallback(
+    (themeLabel: ThemeLabel) => {
+      dispatch({
+        type: 'set_Theme',
+        themeLabel: themeLabel,
+      });
+    },
+    [dispatch],
+  );
+
+  // True to show the settings modal.
+  const setIsSettingsOpen = useCallback(
```

Comment:
Excellent collection of functions that dispatch actions!

I could imagine moving all of these into a separate module.

Similar pattern is here from the old days of Redux: https://react-redux.js.org/using-react-redux/connect-mapdispatch